packages/runner/lib/sdk/sdk.ts

```diff
@@ -381,12 +382,55 @@ export class NangoSyncRunner extends NangoSyncBase {
         }
         return true;
     }
+
+    public async getRecordsByIds<K = any, T = any>(ids: K[], model: string): Promise<Map<K, T>> {
```

Comment:
should we limit the number of records that can be requested at once. As it is a customer can fetch an unlimited number of records

Reply:
also would it nicer to make it a generator so customer can paginate over the result?

Reply:
I could see doing a generator on a record by record basis so that they're minimizing the resident set. Good call. I'm thinking it would look like:

```
for await (const record of nango.getRecordsByIds(['a', 'b', 'c'], 'Record')) {
  record.newKey = 'biz';
  nextBatch.push(record);
}
```

Reply:
I'm not that concerned about allowing unlimited keys. Customers could technically also write a tight loop that pushes into an array that would cause an OOM, but at the end of the day I'm not sure without us building a memory safe language that we can avoid all abuse possibilities.